src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -205,4 +212,48 @@ public Coords getComputerFireCoords() {
 
         return new Coords(across, down);
     }
+
+    public void placeAIShip(String name){
+        String orient;
+        int dir;
+        boolean fail;
+        Random randNum = new Random();
+
+
+        do {
+            int x = (randNum.nextInt(GRID_SIZE) + 1);
```

Comment:
This approach might run into trouble, as Ships will extend either to the right or down depending on their orientation, and this has the potential to put them so close to the grid's edges that the ship would be partially off-grid. You'll probably want to check if a ship will be off-grid (or whether updatePosition successfully places it or not) and re-randomize as needed, or handle it some other way.